Formula/g/gnome-builder.rb

```diff
@@ -0,0 +1,118 @@
+  depends_on "libadwaita"
+  depends_on "libgit2-glib"
+  depends_on "libpanel"
+  depends_on "libpeas"
+  depends_on "llvm"
+  depends_on "template-glib"
+  depends_on "vte3"
+
+  resource "libdex" do
+    url "https://download.gnome.org/sources/libdex/0.7/libdex-0.7.0.tar.xz"
+    sha256 "ee789f308e8edcd7dd24553bbb543f4b8e37aee64d0589936fa22b9d1bf69f15"
+
+    # macOS ucontext fix, remove in next version
+    patch do
+      url "https://gitlab.gnome.org/GNOME/libdex/-/commit/fb8727c2dee6c9636517d5e7673976621894ae5b.diff"
+      sha256 "cd8887115774aed0780b5b9896b2a93493a53b6e914937ec92ead90cd5de3b04"
+    end
+  end
+
+  # Fixes compilation failure on Linux. See:
+  # https://gitlab.gnome.org/GNOME/gnome-builder/-/issues/2176
+  patch do
+    on_linux do
+      url "https://gitlab.gnome.org/GNOME/gnome-builder/uploads/3595cbe09fb9dd41d558a317251ca356/gdkrgba.patch"
+      sha256 "5234eb22edba53a6977cd1ed9c0de667115c65bb25ddb3ce3cd7cb43cdd7592f"
+    end
+  end
+
+  # Prevents Meson from compiling schemas / updating icon cache separately from
+  # other GTK programs. We want to compile schemas / update icon cache all
+  # together
+  patch "diff --git a/meson.build b/meson.build
+index e486d52..39afd1d 100644
+--- a/meson.build
++++ b/meson.build
+@@ -388,8 +388,8 @@ subdir('po')
+ subdir('doc')
+
+ gnome.post_install(
+-  glib_compile_schemas: true,
+-  gtk_update_icon_cache: true,
++  glib_compile_schemas: false,
++  gtk_update_icon_cache: false,
+   update_desktop_database: true,
+ )"
+
+  def install
+    resource("libdex").stage do
+      system "meson", "setup", "build", *std_meson_args
+      system "meson", "compile", "-C", "build", "--verbose"
+      system "meson", "install", "-C", "build"
+
+      ENV.prepend_path "PKG_CONFIG_PATH", lib/"pkgconfig"
+      ENV.prepend_path "GI_GIR_PATH", share/"gir-1.0"
+    end
+
+    args = %w[
+      -Dplugin_dspy=false
+      -Dplugin_flatpak=false
+      -Dplugin_html_preview=false
+      -Dplugin_markdown_preview=false
+      -Dplugin_sphinx_preview=false
+      -Dplugin_update_manager=false
+      -Dwebkit=disabled
+    ]
+
+    # This plugin fails to compile in 46.2, should be fixed in later versions
+    args << "-Dplugin_git=false" unless build.head?
+    # This plugin does not exist in 46.2, but on head it does and requires
+    # WebKit
+    args << "-Dplugin_manuals=false" if build.head?
+
+    system "meson",
+      "setup",
+      "build",
+      *args,
+      *std_meson_args
+    system "meson", "compile", "-C", "build", "--verbose"
+    system "meson", "install", "-C", "build"
+  end
+
+  # See patch comment on line 87
+  def post_install
+    system Formula["glib"].opt_bin/"glib-compile-schemas", HOMEBREW_PREFIX/"share/glib-2.0/schemas"
+    system Formula["gtk4"].opt_bin/"gtk4-update-icon-cache", "-qtf", HOMEBREW_PREFIX/"share/icons/hicolor"
+  end
+
+  test do
+    assert_equal "GNOME Builder #{version}", shell_output("#{bin}/gnome-builder --version").strip
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
As far as I'm aware, all command line options open the GUI aside from things like `--version` and `--help`